src/ng_package/packager/index.ts

```diff
@@ -77,16 +77,32 @@ function main(args: string[]): void {
     esm2022Arg,
 
     // List of static files that should be copied into the package.
-    staticFilesArg,
+    staticLocationsArg,
 
     // List of side-effectful entry-points
     sideEffectEntryPointsArg,
   ] = params;
 
   const esm2022 = JSON.parse(esm2022Arg) as BazelFileInfo[];
-  const staticFiles = JSON.parse(staticFilesArg) as BazelFileInfo[];
+  const staticLocations = JSON.parse(staticLocationsArg) as BazelFileInfo[];
   const metadata = JSON.parse(metadataArg) as PackageMetadata;
   const sideEffectEntryPoints = JSON.parse(sideEffectEntryPointsArg) as string[];
+  const staticFiles: BazelFileInfo[] = [];
+
+  // Investigate each location provided, and expact directory locations into all of the files
+  // within the directory.
+  for (const location of staticLocations) {
+    if (fs.lstatSync(location.path).isDirectory()) {
```

Comment:
OOC, why aren't we simply using the `nested_packages` logic here?

Reply:
Thats what we had been doing, but then we aren't using it for actual nested packages, we are just using it as a cpoy to the npm package in place.  Which functionally is fine, but just felt wrong.   If you would prefer we do it that way, we can close this and continue doing that.

Reply:
I think it's fine to continue doing that. We would just use a `js_library`, or leverage e.g. `copy_to_directory` (from aspect bazel lib).

Can you give an example where this is needed? Would love to only have a single way (npm_package + replace_prefixes)